test/integration/deploy_trigger_test.go

```diff
@@ -192,14 +194,18 @@ waitForNewConfig:
 		case event := <-configWatch.ResultChan():
 			if event.Type == watchapi.Modified {
 				newConfig = event.Object.(*deployapi.DeploymentConfig)
-				break waitForNewConfig
+				// Multiple updates to the config can be expected (e.g. status
+				// updates), so wait for a significant update (e.g. version).
+				if newConfig.LatestVersion > 0 {
```

Comment:
Haven't found anything yet, will make followups as necessary. I'd like to knock this one out now.